cmd/metrics.go

```diff
@@ -64,6 +67,30 @@ func (c *minioCollector) Describe(ch chan<- *prometheus.Desc) {
 // Collect is called by the Prometheus registry when collecting metrics.
 func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
 
+	globalMetricsChan<- 1
+	// Waiting to make sure metrics thread finishes before
+	<-globalMetricsChan
+	ch <- prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName("minio", "metrics", "iops"),
+			"Current IOPS of this Minio instance",
+			nil, nil),
+		prometheus.CounterValue,
+		float64(atomic.LoadUint64(&globalCurrIOCount)),
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName("minio", "metrics", "bw"),
+			"Current BW of this Minio instance (KiB/s)",
+			nil, nil),
+		prometheus.CounterValue,
+		float64(float64(atomic.LoadUint64(&globalCurrBW)) / float64(1024)),
+	)
+	// Reset counters once reported
+	atomic.StoreUint64(&globalCurrIOCount, 0)
+	atomic.StoreUint64(&globalCurrBW, 0)
+	
```

Comment:
Lets discuss, the reason I have an issue with this design as metric collection is triggered by this function. Now, metrics should be independent to Prometheus..For example, if there is a request to get metrics via ustat, then you need to change the design..Rather what I suggested earlier from the  metricsReporting() thread, you reset the counter every 1 sec and keep on collecting if a global variable like metric_collection is enabled during startup...The downside of that approach is, we keep on collecting the metrics till lifetime of minio..Lets discuss more..